superset/dashboards/filter_state/commands/create.py

```diff
@@ -23,13 +25,17 @@
 
 
 class CreateFilterStateCommand(CreateKeyValueCommand):
-    def create(self, cmd_params: CommandParameters) -> bool:
+    def create(self, cmd_params: CommandParameters) -> str:
         resource_id = cmd_params.resource_id
         actor = cmd_params.actor
-        key = cache_key(resource_id, cmd_params.key)
+        contextual_key = cache_key(actor.get_user_id(), resource_id)
+        key = cache_manager.filter_state_cache.get(contextual_key)
+        if not key:
+            key = token_urlsafe(48)
```

Comment:
This part I find problematic. I get the idea that we want to avoid creating new keys every time we load the same resource with the same key. But recycling the same key every time the same user navigates to the same resource is IMO slightly risky. I'm sure it's a fairly common use case to be exploring the same resource in multiple different tabs. Making them all share the same key-value pair will make it impossible to do this type of exploration without leaking state between the tabs. I tried to record a video to demonstrate what I mean, but here's the repro steps:
1. Open Superset in two tabs
2. Navigate the first tab to your favourite dashboard and set a few native filters
3. Navigate the second tab to the same dashboard
4. refresh the first tab and notice that the native filter state has been lost

Reply:
Nice point. I'll work on it and submit a new commit supporting this.